Add X-Frame-Options to page headers

emrysal opened the pull request:
## What does this PR do?

Uses regex to check if a page ends with /embed, if not we set the embed policy to DENY.

emrysal committed:
Add X-Frame-Options to /auth/login headers

emrysal committed:
Don't allow embedding any page except a page that ends with /embed

hariombalhara commented on apps/web/next.config.js:
```diff
@@ -176,6 +176,19 @@ const nextConfig = {
+  async headers() {
+    return [
+      {
+        source: "/:path*((?<!embed)$)",
```
```suggestion
        source: "/:path*((?<!(embed|embed\/preview.html))$)",
```

hariombalhara committed:
Improve RegEx to fix 2 cases

- myembed should be DENYd but my/embed shouldn't be
- preview.html needs whitelisting too.

hariombalhara committed:
Update next.config.js

emrysal committed:
Disable prettier, we know better

emrysal committed:
Different method of prettier-ignore + seperate negative lookaheads
